website/docs/migration.md

```diff
@@ -0,0 +1,244 @@
+   In the majority of cases, this does not necessitate additional scrutiny of the diff output.
+   Nevertheless, if FerretDB does not handle the error, additional inspection becomes necessary.
+
+   ```bash
+   # run mongosh
+   me@foobar:~/FerretDB$ mongosh --quiet
+   ```
+
+   ```js
+   // insert some documents
+   test> db.posts.insertMany([
+   ...   {
+   ...     title: 'title A',
+   ...     body: 'some content',
+   ...     author: 'Bob',
+   ...     date: ISODate("2023-08-29T10:33:23.134Z"),
+   ...   },
+   ...   {
+   ...     title: 'another title',
+   ...     body: 'some content',
+   ...     author: 'Bob',
+   ...     date: ISODate("2023-08-28T10:33:23.134Z"),
+   ...   },
+   ...   {
+   ...     title: 'title B',
+   ...     body: 'some content',
+   ...     author: 'Alice',
+   ...     date: ISODate("2023-08-20T10:33:23.134Z"),
+   ...   },
+   ...   {
+   ...     title: 'some other title',
+   ...     body: 'some content',
+   ...     author: 'Alice',
+   ...     date: ISODate("2023-08-21T10:33:23.134Z"),
+   ...   },
+   ... ]);
+   {
+     acknowledged: true,
+     insertedIds: {
+       '0': ObjectId("64edcfc975dd10e7bfb36add"),
+       '1': ObjectId("64edcfc975dd10e7bfb36ade"),
+       '2': ObjectId("64edcfc975dd10e7bfb36adf"),
+       '3': ObjectId("64edcfc975dd10e7bfb36ae0")
+     }
+   }
+   test> // run a query to fetch the first post from each author sorted by date and author
+   test> db.posts.aggregate(
+   ...   [
+   ...     { $sort: { date: 1, author: 1 } },
+   ...     {
+   ...       $group:
+   ...         {
+   ...           _id: "$author",
+   ...           firstPost: { $first: "$date" }
+   ...         }
+   ...     }
+   ...   ]
+   ... );
+   MongoServerError: $group accumulator "$first" is not implemented yet
+   test>
+   ```
+
+### Manual and automated testing with `diff-proxy` mode
+
+`diff-proxy` mode will forward client requests to a MongoDB-compatible database, but return the proxy responses to the client, and will log the difference between them.
+
+Continuing with the same example above, we can further examine the diff output.
+
+1. Run FerretDB in `diff-proxy` mode.
+
+   ```bash
+   me@foobar:~/FerretDB$ bin/task run-proxy
+   ```
+
+2. Re-run the query.
+
+   ```js
+   test> db.posts.aggregate(
+   ...   [
+   ...     { $sort: { date: 1, author: 1 } },
+   ...     {
+   ...       $group:
+   ...         {
+   ...           _id: "$author",
+   ...           firstPost: { $first: "$date" }
+   ...         }
+   ...     }
+   ...   ]
+   ... );
+   [
+     { _id: 'Alice', firstPost: ISODate("2023-08-20T10:33:23.134Z") },
+     { _id: 'Bob', firstPost: ISODate("2023-08-28T10:33:23.134Z") }
+   ]
+   test>
+   ```
+
+In the diff output below we have discovered that the query cannot be serviced by our application because the `$first` accumulator operator is not implemented in FerretDB.
+
+```bash
```

Comment:
```suggestion
```diff
```

Reply:
Wasn't aware of `diff`.

Reply:
It is used on https://docs.ferretdb.io/configuration/operation-modes/#diff-modes